src/cplus_plugin/api/scenario_task_api_client.py

```diff
@@ -258,6 +249,14 @@ def upload_layers(self):
             Settings.SIEVE_ENABLED, default=False, setting_type=bool
         )
 
+        priority_layers = settings_manager.get_priority_layers()
+        for priority_layer in priority_layers:
+            if priority_layer['uuid'] in activity_pwl_uuids and os.path.exists(priority_layer['path']):
```

Comment:
I think it's better to use priority_layer.get so we don't get key error exception. The original code in tasks.py is also using .get('").
Please apply the same when retrieving 'groups' and 'value'.